### PR TITLE
build(version): build image as quay.io/cryostat/cryostat with lowercase version tag

### PR DESCRIPTION
**Pull request by andrewazores**
# Welcome to Cryostat3! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat3/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits using a GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification)

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Fixes: #124

### CODE REVIEW

**Comment by andrewazores** [comment]
Sample run on my fork:

https://github.com/andrewazores/cryostat3/actions/runs/6789839546/job/18457931263

Failed pushing to quay.io as expected, since my fork doesn't have the right secrets.

File: .github/workflows/push-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        podman rmi ${{ env.CI_IMG }}:latest ${{ env.CI_IMG }}:dev ${{ env.CI_IMG }}:${{ env.IMAGE_VERSION }}
+      continue-on-error: true
+    - name: Build container images and manifest
+      if: ${{ matrix.java == '17' }} && ${{ github.repository_owner == 'cryostatio' }}
+      id: buildah-build
+      uses: redhat-actions/buildah-build@v2
+      with:
+        image: ${{ env.CI_IMG }}
+        archs: amd64, arm64
+        # tags: ${{ env.IMAGE_VERSION }} ${{ github.ref == 'refs/heads/main' && 'latest' || '' }}
```
[the remark this answers is not in the document]
Comment (andrewazores) [reply .github/workflows/push-ci.yaml:120]: I can remove it for now, but once we make 3.0 the next targeted release I think we would put this (or something similar) back. Right now I have set this up to only push to quay.io with the `3.0.0-snapshot` tag, and `latest` would actually still point to whatever the latest snapshot is from the old repo.

File: .github/workflows/push-ci.yaml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      - cryostat-v[0-9]+.[0-9]+
+
+env:
+  OPENSUSE_UNOFFICIAL_LIBCONTAINERS_KEY_URL: "https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04/Release.key"
+  OPENSUSE_UNOFFICIAL_LIBCONTAINERS_SOURCE_URL: "https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04"
+  CI_USER: cryostat+bot
+  CI_REGISTRY: quay.io/cryostat
+  CI_IMG: quay.io/cryostat/cryostat
+  REPOSITORY: ${{ github.event.pull_request.head.repo.full_name }}
+  REF: ${{ github.event.pull_request.head.ref }}
```
[the remark this answers is not in the document]
Comment (andrewazores) [reply .github/workflows/push-ci.yaml:22]: Ah, I think I can remove them here.

**Comment by andrewazores** [comment]
New test run: https://github.com/andrewazores/cryostat3/actions/runs/6790922942

File: .github/workflows/push-ci.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ebaron) [reply .github/workflows/push-ci.yaml:116]: Sounds good. No need to remove it.

**Review by ebaron — APPROVED** [review]

**Pull request merged**